engine/src/flutter/shell/platform/android/flutter_main.cc

```diff
@@ -230,6 +232,11 @@ bool FlutterMain::Register(JNIEnv* env) {
   return env->RegisterNatives(clazz, methods, std::size(methods)) == 0;
 }
 
+// static
+bool FlutterMain::IsDeviceEmulator(std::string_view product_model) {
+  return std::string(product_model).find("gphone") != std::string::npos;
```

Comment:
Is there ever a case this could also just be "emulator"?

Reply:
probably, but I decided to play it conservative since all my emulators said gphone. (We only have to worry about very new emulators as old ones have no vulkan support regardless of API level)